changesets/scripts/entrypoint.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+search_key=""
+search_value=""
+cached=false
+modify_string="hgv=no to hgv=yes"
+
+escape_spaces() {
+  echo "${1// /\\ }"
+}
+
+USAGE_STRING="Correct usage: generate_changeset.sh --search-key \"<search key>\" --search-value \"<search value>\""
+OPTSTRING="k:v:c"
+LONGOPTSTRING="search-key:,search-value:,cached"
+
+if ! ARGS=$(getopt -o "$OPTSTRING" --long "$LONGOPTSTRING" -- "$@"); then
+  echo "Failed to parse args"
+  echo "$USAGE_STRING"
+  exit 1
+fi
+
+eval set -- "$ARGS"
+
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    -k|--search-key)
+      shift
+      search_key=$(escape_spaces "$1")
+      shift 
+      ;;
+    -v|--search-value)
+      shift
+      search_value=$(escape_spaces "$1")
+      shift
+      ;;
+    -c|--cached)
+      shift
+      cached=true
+      shift
+      ;;
+    --)
+      shift
+      break
+      ;;
+    *)
+      echo "Unknown option: $1"
+      echo "$USAGE_STRING"
+      exit 1
+      ;;
+  esac
+done
+
+if [[ -z "$search_key" ]]; then
+  echo "Search key not found. Please provide in the format of -k \"<search key>\" or --search-key \"<search key>\"";
+  exit 1
+fi
+
+if [[ -z "$search_value" ]]; then
+  echo "Search value not found. Please provide in the format of -v \"<search value>\" or --search-value \"<search value>\"";
+  exit 1
+fi
+
+if [ "$cached" = true ]; then
```

Comment:
question: is there a reason to use `[` here and not `[[` like the above?

```suggestion
if [[ "$cached" = true ]]; then
```

Reply:
no, just an oversight. updated